plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -456,36 +459,61 @@ public DeltaLakeMetadata(
         this.allowManagedTableRename = allowManagedTableRename;
     }
 
-    public TableSnapshot getSnapshot(ConnectorSession session, SchemaTableName table, String tableLocation, long atVersion)
+    public TableSnapshot getSnapshot(ConnectorSession session, SchemaTableName table, String tableLocation, Optional<Long> atVersion)
     {
-        return getSnapshot(session, table, tableLocation, Optional.of(atVersion));
-    }
+        if (atVersion.isEmpty()) {
+            atVersion = Optional.ofNullable(queriedVersions.get(new QueriedTable(table, Optional.empty())));
```

Comment:
I don't like the fact that we are modyfing this variable. Can we do it something like:

```
Optional<Long> version = atVersion.or(() -> Optional.ofNullable(queriedVersions.get(new QueriedTable(table, Optional.empty()))));
```